### PR TITLE
[Merged by Bors] - chore(Pointwise): Rename `Set` to `Set.Basic`

### PR DESCRIPTION
**Pull request by YaelDillies**
Move:
* `Algebra.Group.Pointwise.Set` → `Algebra.Group.Pointwise.Set.Basic`
* `Algebra.Group.Pointwise.Finset.NatCard` → `Algebra.Group.Pointwise.Set.Card`
* `Algebra.GroupWithZero.Pointwise.Set` → `Algebra.GroupWithZero.Pointwise.Set.Basic`

This is making space for a new file `Algebra.GroupWithZero.Pointwise.Set.Card` in #17231.

`Algebra.Group.Pointwise.Finset` already has this `.Basic`  naming scheme.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Order/LeftRightNhds.lean [hunk 1]
```diff
@@ -3,8 +3,7 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Mario Carneiro, Yury Kudryashov
 -/
-import Mathlib.Algebra.Ring.Pointwise.Set
-import Mathlib.Order.Filter.AtTopBot.Group
+import Mathlib.Algebra.Ring.Pointwise.Set\nimport Mathlib.Order.Filter.AtTopBot.Group
```
Comment (b-mehta) [inline Mathlib/Topology/Order/LeftRightNhds.lean:6]:
```suggestion
import Mathlib.Algebra.Ring.Pointwise.Set.Basic
import Mathlib.Order.Filter.AtTopBot.Group
```

**Comment by eric-wieser** [comment]
Can you be explicit in the description about precisely what the name is, and what the file it is making room for is called?

**Comment by eric-wieser** [comment]
I sort of dispute the claim that space needs to be made here. Why can't we have:

* `Algebra/Group/Pointwise`
  * `Set.lean`
  * `Set/Card.lean`

**Comment by eric-wieser** [comment]
Ah, I guess this ship has locally sailed with the adjacent `Finset`, so I'll let you have this for local consistency

bors d+

**Comment by YaelDillies** [comment]
Is there any downside to renaming? Historically, every time we have wanted to add a second file similar to file `X` we've renamed `X` to `X.Basic` and added `X.NewFile`. Note that it also matches the `Pointwise.Finset` folder structure.

**Comment by eric-wieser** [comment]
> Historically, every time we have wanted to add a second file similar to file `X` we've renamed `X` to `X.Basic` and added `X.NewFile`.

I claim we should stop doing this, it breaks downstream users for no good reason. But indeed, for this PR the other argument is enough to sway me.

**Comment by YaelDillies** [comment]
> I claim we should stop doing this, it breaks downstream users for no good reason.

Historically, those folders basically never shrink after they are created, and it doesn't really make sense to me that we should have `.Basic` files iff the original author planned that the subject would extend to more than one file.

This organic growth of folders is one of the main attrayals of mathlib IMO. I don't see why we should get rid of it.

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**